src/modules/esl-sortable/core/esl-sortable.ts

```diff
@@ -0,0 +1,318 @@
+    this._$targetEl = targetMixin;
+    this.targetSortable = targetMixin.parent;
+    this.fireInsertEvents(false);
+  }
+
+  private fireInsertEvents(isSortableEmpty: boolean = false): void {
+    this._$targetEl.$$cls('esl-sortable-target', true);
+    if (!isSortableEmpty && !this.isSortableFromGroup(this.targetSortable)) return;
+    const insertData = {detail: {target: this._$targetEl, placeholder: this._$placeholderEl, sameSortable: isSortableEmpty}};
+    this.placeholderSortable.$$fire('sortable:insert', insertData);
+    // Fired when element was dragged from one sortable to another
+    if (this.targetSortable !== this.placeholderSortable) {
+      this.targetSortable.$$fire('sortable:insert', insertData);
+    }
+  }
+
+  /** Checks if target is sortable */
+  private isSortable(el: HTMLElement): boolean {
+    return (el.constructor as typeof ESLBaseElement).is === 'esl-sortable';
+  }
+
+  /** Checks if targeted sortable is empty */
+  private isSortableEmpty(el: ESLSortable): boolean {
+    return this.isSortable(el) ? el.childrens.length === 0 : false;
+  }
+
+  /** Checks if targeted sortable is from the same group */
+  private isSortableFromGroup(el: ESLSortable): boolean {
+    return this.isSortable(el) ? el.group === this.group : false;
+  }
+
+  @bind
+  protected _onInsert(e: CustomEvent): void {
+    const {target, placeholder, sameSortable} = e.detail;
+    this._$targetEl = target;
+    this._$placeholderEl = placeholder;
+    this.targetSortable = sameSortable ? target : target.parent;
+    this.placeholderSortable = placeholder.parent;
+    this.insertChild();
+  }
+
+  /** Inner state */
+  private get transition(): boolean {
+    return this._transition;
+  }
+  private set transition(value: boolean) {
+    this.toggleAttribute('transition', this._transition = value);
+  }
+
+  /** Inner state */
+  private get empty(): boolean {
+    return this._empty;
+  }
+  private set empty(value: boolean) {
+    this.classList.toggle('esl-sortable-empty', this._empty = value);
+  }
+
+  /** Method recalculates sortable's children position two times: before and after element insert */
+  private insertChild(): void {
+    this.targetSortable.transition = true;
+    this.placeholderSortable.transition = true;
+    this.positionItems();
+    afterNextRender(() => {
+      if (this._$targetEl) this._$targetEl.$$cls('esl-sortable-target', false);
+      // Necessary if element was transfered from one sortable to another
+      if (this._$placeholderEl.$$cls('esl-sortable-group-inserted')) {
+        this.positionItems(true);
+        return;
+      }
+
+      if (this.targetSortable.empty) {
+        this.targetSortable.empty = false;
+        this.targetSortable.list.appendChild(this._$placeholderEl.$host);
+      } else {
+        (this._$targetEl as ESLSortableItem).$host.insertAdjacentElement(this.isInsertedBefore() ? 'afterend' : 'beforebegin', this._$placeholderEl.$host);
+      }
```

Comment:
Looks like can be delegated with possible improvement for single responsibility term